libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/router/RouterInterfaceAdapter.kt

```diff
@@ -126,6 +127,14 @@ class RouterInterfaceAdapter(
         return requestId
     }
 
+    override fun getRoute(
+        directionsUri: String,
+        options: GetRouteOptions,
+        callbackDataRef: RouterDataRefCallback
+    ): Long {
+        throw UnsupportedOperationException()
```

Comment:
That could cause a crash in case user provides custom router and NN uses get route with `data_ref` internally.
@mapbox/navnative , do you use this method internally?

Reply:
BTW, give the fact that we're not aware of such users, maybe it's okay to return an error in the callback every time just to avoid crash?

Reply:
Actually, it's not hard to implement it, see https://github.com/mapbox/mapbox-java/pull/1526/files#diff-cbe0d3a161605b88f1f3fa4350086fd8ea0530e774da3737ddfacc343bec9823R101 as an example to convert string to butebuffer

Reply:
I don't want to do it here unless it's absolutely necessary. 
1. We don't expose NN API, so `RouterInterface` object can't be accessed by the user, so they can't access the method themselves.
2. NN might invoke it, as you pointed out, so I'll confirm with them they are not using it. Otherwise I'll have to adapt.

Reply:
> I don't want to do

I don't mind keeping it simple and adopt in the followup PR(probably I will do it). Maybe you are okay to replace exception which crashes an app with `callbackDataRef.onError("not supported yet")`, so that even if somebody uses custom router, the SDK at least doesn't crash the app.

Reply:
Done.